workflow-controller/controller-service/src/main/java/io/hyscale/controller/util/ServiceProfileUtil.java

```diff
@@ -60,6 +65,70 @@
         return serviceProfileMap;
     }
 
+    /**
+     *  Looks for profile files in spec as well as profiles directory.
+     * @param serviceSpecs
+     * @param profileName
+     * @return
+     * @throws HyscaleException if profile found at both the places, or no profile found
+     */
+    public static List<File> getProfilesFromName(List<File> serviceSpecs, String profileName) throws HyscaleException {
+        if (serviceSpecs == null || StringUtils.isBlank(profileName)) {
```

Comment:
Why filter them before effective service spec?
get all of them found with the profile name and later filter by services accordingly.

Reply:
In that case we will end up validating profiles which are not used. Lets discuss more on this.